%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%                                                              %%%%%
%%%%% Chapter 2: Background                                        %%%%%
%%%%%                                                              %%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter{Background}
\label{ch2}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Nanopore sequencing}
%% Intro

%%% History of nanopore development
\paragraph{A brief history of nanopore sequencing}
%% Initial idea and expreiments
The concept of nanopore sequencing is based on the idea that as a
single-stranded DNA (or RNA) translocates through a nanopore (such as
transmemberane protein) in the presence of an electric field, the change
in current level would be proportional to the nucleotide passing through
the nanopore; Thus, measuring the current over time could be leveraged
to determine the sequence of nucleotides.
%
This idea of using transmemberane proteins for sensing and sequencing
nucleic acids was independently thought of by several researches
including David Deamer, Hagan Bayley, and George Church
\citep{deamer2016three,bayley2015nanopore,branton2010potential}.

%% Initial expreiments
% Detecting the presence of oligos
Initial experiments showed that as a single-stranded DNA or RNA
molecules could be driven through a \emph{Staphylococcus aureus}
$\alpha$-hemolysin in the presence of an electric field
\citep{kasianowicz1996characterization}. The current through the pore
remained constant in the absence of oligomers; The presence of oligomers
caused transient decreases in current, with the duration of the decrease
proportional to the length of the oligomer.
% Detecting the bases in oligos
Further research demonstrated that decrease in amplitude of current
could be used to different between poly-purine and poly-pyrimidine
sequencing of RNA \citep{akeson1999microsecond} and DNA
\citep{meller2000rapid}.

%% Challenges for sequencing DNA
Although these experiments demonstrated the potential for nanopores to
distinguish nucleic acid polymers, several challenges remained to be
addressed to use this approach for reading individual bases on a DNA or
RNA molecules.
%
The most important of these challenges were...


%% Capture rate and fractors that affect it
\paragraph{Factors that affect nanopore sequencing}

%% Oxford nanopore technologies
\paragraph{Oxford Nanopore Technologies}

%% Oxford MinION


%% Current state of nanopore sequencing



%% Nanopore library construction
\paragraph{Nanopore library preparation and sequencing}
%% TODO: is it called library preparation or sample preparation
Sequencing nucleic acids requires prepossessing the sample DNA for
compatibility with the underlying sequencing technology, a process
traditionally referred to as library preparation.
%
Sequencing on a nanopore machine usually requires fragmenting DNA
molecules to the appropriate length and attaching sequencing adapters.
%
Oxford nanopore technologies offers several commercially library
preparation kits for both DNA and RNA samples.  The most frequently used
of these kits are the Ligation Sequencing Kit family and the Rapid
Sequencing Kit family.

% Ligation kit
In theory, there is no limit on the length of a molecule that can be
sequenced with a nanopore, and thus, the length is determined by the
downstream application or the limitations of handling high molecular
weight DNA. For the ligation sequencing kit (), the recommended length is
$\sim$8 kb when starting with 1ug of sample for appropriate molar
concentration in the subsequent steps.  DNA molecules can be fragmented
to the appropriate length using a variety of methods including the g-tube
(). %
These molecules are then optionally repaired to remove any nicks, and
then the DNA ends are prepared to have a dA tail. Finally, sequencing
adapters (that have a dT tail) are ligated to the end-prepared DNA.
These adapters contain specific DNA sequenced with attached enzymes
that regulated translocation of the DNA molecule into a nanopore.
Library preparation with the ligation kit takes approximately 60 to
90 minutes.

% Rapid kit
The rapid library preparation kit offers a faster method, by
simultaneously fragmenting and tagging the ends of high molecular weight DNA
(recommended $>$ 30 kb). Adapters are then attached to these tags.
Library preparation with the rapid kit takes approximately 10 minutes.

% Barcoding capablities
Both of these kits offer barcoding capabilities for multiplexing several
samples in a single sequencing run.

% Brief summary of other kits


% After lib. prep.: loading and sequencing

% Base-calling reads


%% Properties of nanopore reads
\paragraph{Properties of nanopore reads}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Copy number variation and profiling}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Prior protocols based on concatenating DNA molecules}
%%% SAGE
\paragraph{Serial analysis of gene expression (SAGE)}
The concept of ligating short DNA molecules to improve the efficiency
of sequencing was introduced in serial analysis of gene expression
(SAGE) \citep{}, and subsequently its variants such as LongSAGE and
SuperSAGE \citep{}. SAGE

%%% variants of SAGE
\paragraph{Variants of SAGE}

% Digital karyotyping
\paragraph{Digital karyotyping}

% SMASH
\paragraph{SMASH}

% concat-seq
\paragraph{Concat-seq}

